overlord/hookstate/hooktest/handler.go

```diff
@@ -41,6 +45,10 @@ func NewMockHandler() *MockHandler {
 
 // Before satisfies hookstate.Handler.Before
 func (h *MockHandler) Before() error {
+	executing := atomic.AddInt32(&h.Executing, 1)
+	if executing != 1 {
+		panic(fmt.Sprintf("More than one handler executed: %d", executing))
```

Comment:
Added second test, both tests should now cover all cases (making sure we serialize hook tasks for same snap, and making sure we allow hooks from different snaps run concurrently).